### PR TITLE
Added gzip example

### PR DESCRIPTION
**Pull request by dgrr**

### CODE REVIEW

**Comment by erikdubbelboer** [comment]
Thank you.

File: examples/gzipped/main.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	"log"
+
+	"github.com/erikdubbelboer/fasthttp"
+)
+
+func server() {
+	server := fasthttp.Server{
+		Name:              "Fasthttp server",
+		Handler:           handler,
+		ReduceMemoryUsage: true,
```
Comment (thehowl) [inline examples/gzipped/main.go:15]: Why is this enabled?

File: examples/gzipped/main.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	}
+	log.Fatal(server.ListenAndServe(":1313"))
+}
+
+func handler(ctx *fasthttp.RequestCtx) {
+	ctx.SetContentType("text/html")
+	ctx.Response.Header.Add("Content-Encoding", "gzip")
```
Comment (thehowl) [inline examples/gzipped/main.go:22]: We should not say that the Content-Encoding is gzip when we don't even know the client will accept gzip! (Move this one line below)

File: examples/gzipped/main.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	req.Header.Add("Accept-Encoding", "gzip")
+	req.SetRequestURI("http://localhost:1313")
+
+	err := fasthttp.Do(req, res)
+	if err != nil {
+		panic(err)
+	}
+	body := res.Body()
+	if b := res.Header.Peek("Content-Encoding"); len(b) > 0 {
+		if bytes.Index(b, []byte("gzip")) >= 0 {
```
Comment (thehowl) [inline examples/gzipped/main.go:51]: simplify to `if bytes.Equal(res.Header.Peek("Content-Encoding"), []byte("gzip"))`?

**Review by thehowl — COMMENTED** [review]
Sorry for the late review - currently going through every PR and issue in my notifs for this repo and they've stacked up over time. A few minor issues, but I still think you should send another PR to address these.

File: examples/gzipped/main.go (hunk 1 above)
Comment (dgrr) [reply examples/gzipped/main.go:15]: Programming habit xd

**Pull request merged**